src/core/authenticator.cpp

```diff
@@ -27,6 +27,12 @@ using namespace std;
 Authenticator::Authenticator(const Config &config) {
     mysql_init(&con);
     Log::log_with_date_time("connecting to MySQL server " + config.mysql.server_addr + ':' + to_string(config.mysql.server_port), Log::INFO);
+    if (config.mysql.cafile != "") {
+        mysql_ssl_set(&con, NULL, NULL, config.mysql.cafile.c_str(), NULL, NULL);
```

Comment:
According to https://dev.mysql.com/doc/refman/8.0/en/mysql-ssl-set.html,
```suggestion
        mysql_options(&con, MYSQL_OPT_SSL_CA, config.mysql.cafile.c_str());
```
seems simpler.